python/pyspark/sql/types.py

```diff
@@ -1468,6 +1475,36 @@ def __eq__(self, other: Any) -> bool:
         return type(self) == type(other)
 
 
+class VariantVal:
+    """
+    A class to represent a Variant value in Python.
```

Comment:
Should probably add `    .. versionadded:: 4.0.0` if this is a user-facing API.

Reply:
and preferably examples and paramters like https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
Updated the comments.